# Paulina Lizet Gutiérrez Amezcua A01639948

# Data management using Pandas

## Activity: work with the iris dataset

Repeat this tutorial with the iris data set and respond to the following inquiries

1. Calculate the statistical summary for each quantitative variables. Explain the results
    - Identify the name of each column
    - Identify the type of each column
    - Minimum, maximum, mean, average, median, standar deviation
    
    
2. Are there missing data? If so, create a new dataset containing only the rows with the non-missing data


3. Create a new dataset containing only the petal width and length and the type of Flower


4. Create a new dataset containing only the setal width and length and the type of Flower


5. Create a new dataset containing the setal width and length and the type of Flower encoded as a categorical numerical column 



# Importing libraries


In [119]:
# Import the packages that we will be using
import pandas as pd                 # For data handling

#pd.set_option('display.max_columns', 100) # Show all columns when looking at dataframe

# Importing data

In [120]:
# url string that hosts our .csv file
url = "datasets/iris/iris.csv"

# Read the .csv file and store it as a pandas Data Frame
df = pd.read_csv(url)
df = pd.read_csv(url, names=["sepal length","sepal width","petal length","petal width","class"])

df

sepal length  sepal width  petal length  petal width           class
0             5.1          3.5           1.4          0.2     Iris-setosa
1             4.9          3.0           1.4          0.2     Iris-setosa
2             4.7          3.2           1.3          0.2     Iris-setosa
3             4.6          3.1           1.5          0.2     Iris-setosa
4             5.0          3.6           1.4          0.2     Iris-setosa
..            ...          ...           ...          ...             ...
145           6.7          3.0           5.2          2.3  Iris-virginica
146           6.3          2.5           5.0          1.9  Iris-virginica
147           6.5          3.0           5.2          2.0  Iris-virginica
148           6.2          3.4           5.4          2.3  Iris-virginica
149           5.9          3.0           5.1          1.8  Iris-virginica

[150 rows x 5 columns]

In [121]:
type(df)

pandas.core.frame.DataFrame

### Number of rows and columns

In [122]:
df.shape

(150, 5)

150 (50 in each of three classes)

### Name of each column

In [123]:
df.columns

Index(['sepal length', 'sepal width', 'petal length', 'petal width', 'class'], dtype='object')

4 numeric, predictive attributes and the class

### Type of each column

In [124]:
df.dtypes

sepal length    float64
sepal width     float64
petal length    float64
petal width     float64
class            object
dtype: object

All types in cm.
class: 
      -- Iris Setosa
      -- Iris Versicolour
      -- Iris Virginica

## 1. Minimum, maximum, mean, average, median, standar deviation:

In [125]:
# Summary statistics for the quantitative variables
df.describe()


sepal length  sepal width  petal length  petal width
count    150.000000   150.000000    150.000000   150.000000
mean       5.843333     3.057333      3.758000     1.199333
std        0.828066     0.435866      1.765298     0.762238
min        4.300000     2.000000      1.000000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.350000     1.300000
75%        6.400000     3.300000      5.100000     1.800000
max        7.900000     4.400000      6.900000     2.500000

In [130]:
# Returns the mean of all columns
df.mean()

sepal length    5.843333
sepal width     3.057333
petal length    3.758000
petal width     1.199333
dtype: float64

In [131]:
# Returns the lowest value in each column
df.min()

sepal length            4.3
sepal width             2.0
petal length            1.0
petal width             0.1
class           Iris-setosa
dtype: object

In [132]:
# Returns the highest value in each column
df.max()

sepal length               7.9
sepal width                4.4
petal length               6.9
petal width                2.5
class           Iris-virginica
dtype: object

In [133]:
# Returns the median of each column
df.median()

sepal length    5.80
sepal width     3.00
petal length    4.35
petal width     1.30
dtype: float64

In [134]:
# Returns the standard deviation of each column
df.std()

sepal length    0.828066
sepal width     0.435866
petal length    1.765298
petal width     0.762238
dtype: float64

## 2. Missign Data

In [144]:
df.isnull()

sepal length  sepal width  petal length  petal width  class
0           False        False         False        False  False
1           False        False         False        False  False
2           False        False         False        False  False
3           False        False         False        False  False
4           False        False         False        False  False
..            ...          ...           ...          ...    ...
145         False        False         False        False  False
146         False        False         False        False  False
147         False        False         False        False  False
148         False        False         False        False  False
149         False        False         False        False  False

[150 rows x 5 columns]

None Missing Attribute Values.

## 3. Dataset containing only the petal width and length and the type of Flower:

In [137]:
df_p = pd.read_csv(url)
df.loc[:,["petal length", "petal width", "class"]]

petal length  petal width           class
0             1.4          0.2     Iris-setosa
1             1.4          0.2     Iris-setosa
2             1.3          0.2     Iris-setosa
3             1.5          0.2     Iris-setosa
4             1.4          0.2     Iris-setosa
..            ...          ...             ...
145           5.2          2.3  Iris-virginica
146           5.0          1.9  Iris-virginica
147           5.2          2.0  Iris-virginica
148           5.4          2.3  Iris-virginica
149           5.1          1.8  Iris-virginica

[150 rows x 3 columns]

## 4. Dataset containing only the setal width and length and the type of Flower:

In [138]:
df_s = pd.read_csv(url)
df.loc[:,["sepal length", "sepal width", "class"]]

sepal length  sepal width           class
0             5.1          3.5     Iris-setosa
1             4.9          3.0     Iris-setosa
2             4.7          3.2     Iris-setosa
3             4.6          3.1     Iris-setosa
4             5.0          3.6     Iris-setosa
..            ...          ...             ...
145           6.7          3.0  Iris-virginica
146           6.3          2.5  Iris-virginica
147           6.5          3.0  Iris-virginica
148           6.2          3.4  Iris-virginica
149           5.9          3.0  Iris-virginica

[150 rows x 3 columns]

## Dataset containing the setal width and length and the type of Flower encoded as a categorical numerical column:



In [143]:
df_encoded = pd.read_csv(url, names=["sepal length","sepal width","petal length","petal width","class"])
df_encoded = df_encoded.replace({"class": {"Iris-setosa":0,"Iris-verisicolor":1, "Iris-virginica":2}})
df_encoded

sepal length  sepal width  petal length  petal width class
0             5.1          3.5           1.4          0.2     0
1             4.9          3.0           1.4          0.2     0
2             4.7          3.2           1.3          0.2     0
3             4.6          3.1           1.5          0.2     0
4             5.0          3.6           1.4          0.2     0
..            ...          ...           ...          ...   ...
145           6.7          3.0           5.2          2.3     2
146           6.3          2.5           5.0          1.9     2
147           6.5          3.0           5.2          2.0     2
148           6.2          3.4           5.4          2.3     2
149           5.9          3.0           5.1          1.8     2

[150 rows x 5 columns]